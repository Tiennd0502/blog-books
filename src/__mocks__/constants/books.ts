import { Book } from '@interfaces/book';

const BOOK_ITEM: Book = {
  title: 'The Threshold',
  author: 'Iman Mersal',
  liked: false,
  contributors: ['Iman Mersal', 'Iman Mersal 2', 'Iman Mersal 3'],
  image:
    'https://49thshelf.com/var/ezflow_site/storage/images/books/t/the-threshold/9780374604271_cover/604130664-1-eng-CA/9780374604271_cover_wide_960.jpg',
  summary:
    "A selection of luminous, fiercely intelligent verse from Egypt's premier poet. Iman Mersal is Egypt and indeed the Arab world great outsider poet. Over the past three decades, she has crafted a voice that is ferocious and tender, street-smart and vulnerable. Her early work captures the energies of Cairo's legendary literary boh & Atilde me , peopled by Lovers of hashish and awkward confessions / Anti-state agitators&rdquo; and People like me. These are poems of wit and rage, freaked by moments of sudden beauty, like the scent of guava&rdquo; mysteriously wafting through the City of the Dead. Other poems bear witness to agonizing loss and erotic temptation, the breath of two bodies that never had enough time / and so took pleasure in their mounting terror. Mersal's most recent work addresses itself to the traumas of displacement and migration, as well as the pleasure of crossing boundaries, personal and political, in literature and in life.",
  createdAt: '26-10-2022',
  updatedAt: '26-10-2022',
  id: '1',
};

const BOOK_LIST: Book[] = [
  {
    title: 'The Threshold',
    author: 'Iman Mersal',
    liked: false,
    contributors: ['Iman Mersal', 'Iman Mersal 2', 'Iman Mersal 3'],
    image:
      'https://49thshelf.com/var/ezflow_site/storage/images/books/t/the-threshold/9780374604271_cover/604130664-1-eng-CA/9780374604271_cover_wide_960.jpg',
    summary:
      "A selection of luminous, fiercely intelligent verse from Egypt's premier poet. Iman Mersal is Egypt and indeed the Arab world great outsider poet. Over the past three decades, she has crafted a voice that is ferocious and tender, street-smart and vulnerable. Her early work captures the energies of Cairo's legendary literary boh & Atilde me , peopled by Lovers of hashish and awkward confessions / Anti-state agitators&rdquo; and People like me. These are poems of wit and rage, freaked by moments of sudden beauty, like the scent of guava&rdquo; mysteriously wafting through the City of the Dead. Other poems bear witness to agonizing loss and erotic temptation, the breath of two bodies that never had enough time / and so took pleasure in their mounting terror. Mersal's most recent work addresses itself to the traumas of displacement and migration, as well as the pleasure of crossing boundaries, personal and political, in literature and in life.",
    createdAt: '26-10-2022',
    updatedAt: '26-10-2022',
    id: '1',
  },
  {
    title: 'HARROWINGS',
    author: 'Cecily Nicholson',
    liked: false,
    contributors: [],
    image:
      'https://49thshelf.com/var/ezflow_site/storage/images/books/h/harrowings/9781772014051_cover/629613818-1-eng-CA/9781772014051_cover_wide_960.jpg',
    summary:
      'HARROWINGS takes place mainly in the rural and reconnects with a history of Black intellectual and artistic history in relation to agriculture. The poems include pulses of memoir from the poet&#39;s childhood growing up in the country on a farm. These experiences connect to her volunteer work during the recent pandemic, on a local &#34;prison farm&#34; – an agricultural enterprise whose leadership includes people who were formerly incarcerated. Considering movements organizing for food security, and related, resurgent practices, HARROWINGS addresses the work of cultivation. Underlying references include almanacs and Anglo idioms, drawing upon tabular information, weather, and the workings of the sun, moon, and points of stars as may be practical in relation to a localized, growing year. The poems refuse the romance of husbandry, cultivation, and predictive customs. Understanding &#34;the farm&#34; as a tract of colonial advance – tropes of charming and white, tradition and supremacy, are confronted in a study of biome, water, soil, and seed. With love, despite episodic and chronic illness, duress, and dissociative relationships to time – the poetry advances by way of practical tasks such as watering, weeding, and sowing toward abolitionist futures.',
    createdAt: '26-10-2022',
    updatedAt: '26-10-2022',
    id: '2',
  },
  {
    title: 'Openwork and Limestone',
    author: 'Frances Boyle',
    liked: false,
    contributors: [],
    image:
      'https://49thshelf.com/var/ezflow_site/storage/images/books/o/openwork-and-limestone/9781989466438_cover/617742024-1-eng-CA/9781989466438_cover_wide_960.jpg',
    summary:
      'Openwork and Limestone is a finely-wrought and potent new poetry collection from one of Canada&rsquo;s most compelling poets. In Frances Boyle&rsquo;s powerful vision, the rituals of contemporary women are seen through the lens of Celtic warrior queens, and goddesses. The natural and created worlds &ndash; as they run, as Boyle says, &ldquo;through the funnel / of my palms&rdquo; &ndash; are a constant source of awe and woman&rsquo;s strength. A reverie that allows in the brutality of history and prehistory, as well as the joys. &ldquo;The unconscious / swimming upward. What won&rsquo;t stay buried rises / through rocks, rough-ridden and rusty.&rdquo; Boyle&rsquo;s&nbsp;<em>Openwork and Limestone</em> turns inward and outward at the same time, telling our multifarious collective human story so that it feels like our own intimate family history.',
    createdAt: '26-10-2022',
    updatedAt: '26-10-2022',
    id: '3',
  },
  {
    title: 'Forget-Sadness-Grass',
    author: 'Anthony Di Nardo',
    liked: false,
    contributors: [],
    image:
      'https://49thshelf.com/var/ezflow_site/storage/images/books/f/forget-sadness-grass/9781553806684_cover/629622610-1-eng-CA/9781553806684_cover_wide_960.jpg',
    summary:
      'The sixth collection of poetry from the award-winning poet; Antony Di Nardo takes the reader on a journey into and through mortality, nostalgia, memories of home, lost love, forgetting, the cycle of loss and unexpected yet welcome discovery, with the resilient daylily as an ever present companion; the collection is told with Di Nardos customary grace, modesty, rhythm and precision; poems to learn from, be moved by, surprised by and encouraged by. COMP. TITLES: Skylight, Antony Di Nardo (2018, Ronsdale Press); It Doesnt Matter What We Meant, Rob Winger (2021, Penguin Random House); As Far as You Know, A.F. Mortiz (2020, House of Anansi) Antony Di Nardo&apos;s poems have a sublime precision that relies on the sonic and visual play of language. In poetry that surprises as much as it illuminates, Di Nardo delves into aspects of mortality, memory and forgetting, the cycles of loss and discovery. Di Nardo&apos;s sixth collection takes its name from the Chinese ideogram for daylily, that tawny orange blossom, which bursts its borders in the summer and, in this book, nds its way into poetry that blurs the line between mysticism and lyricism.',
    createdAt: '26-10-2022',
    updatedAt: '26-10-2022',
    id: '4',
  },
  {
    title: 'The Oysters I Bring to Banquets',
    author: 'Gary Geddes',
    liked: false,
    contributors: [],
    image:
      'https://49thshelf.com/var/ezflow_site/storage/images/books/t/the-oysters-i-bring-to-banquets/9781771837101_cover/620754283-1-eng-CA/9781771837101_cover_wide_960.jpg',
    summary:
      'Gary Geddes&rsquo;s new collection of lyrics and poem-sequences ranges from whimsical poems about the building of a greenhouse to the struggle of characters in classical legends to cope with the interference of close relatives and extended family, the gods. And poems about Yukon adventures and the wonders and plight of monarch butterflies in Mexican highlands. It&rsquo;s also a diverse gathering of elegies for friends, literary luminaries, creatures and natural habitats in a world under siege, but also a series of hymns to art, beauty, human dignity and endurance.',
    createdAt: '26-10-2022',
    updatedAt: '26-10-2022',
    id: '5',
  },
  {
    title: 'Dream of Me as Water',
    author: 'David Ly',
    liked: true,
    contributors: [],
    image:
      'https://49thshelf.com/var/ezflow_site/storage/images/books/d/dream-of-me-as-water/9781990293184_cover/620592551-1-eng-CA/9781990293184_cover_wide_960.jpg',
    summary:
      'Moving beyond the themes of race, identity, and personhood navigated in&nbsp;<i>Mythical Man</i>, David Ls second book of poetry,&nbsp;<em>Dream of Me as Water</em>, explores ways of being that are not beholden to the expectations of others. Using water as his central metaphor, Ly meditates on how identity is never a stagnant concept, but instead something that is intangible, fluid, and ever-evolving.&nbsp;<em>Dream of Me as Water</em> revels in the nuances of the self, flouting outside perceptions for deeper, more personal realities. .&nbsp;',
    createdAt: '26-10-2022',
    updatedAt: '26-10-2022',
    id: '6',
  },
];

const BOOK_URL = 'https://61b189803c954f001722a9e5.mockapi.io/books';

const BOOK_LENGTH = BOOK_LIST.length;

const BOOK_ID = BOOK_ITEM.id;

export { BOOK_ITEM, BOOK_LIST, BOOK_URL, BOOK_LENGTH, BOOK_ID };
