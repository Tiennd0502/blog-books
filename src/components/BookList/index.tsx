import React from 'react';
import { SimpleGrid } from '@chakra-ui/react';

import { Book } from '@interfaces/book';
import BookCard from '@components/BookCard';

type Props = {
  books: Book[];
};

const BookList: React.FC<Props> = ({ books }) => {
  return (
    <SimpleGrid
      role="list"
      my="30px"
      columns={{ md: 3, lg: 4, xl: 5 }}
      spacing="30px"
    >
      {books.map((book) => (
        <BookCard key={book.id} book={book} />
      ))}
    </SimpleGrid>
  );
};

export default BookList;
