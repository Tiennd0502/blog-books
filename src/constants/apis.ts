export default {
  BOOK_URL: `${process.env.NEXT_PUBLIC_BASE_URL}/books`,
};
