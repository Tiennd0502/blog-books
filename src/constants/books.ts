export default {
  DETAIL_URL: '/books',
  BOOK_LIMIT: 10,
};
