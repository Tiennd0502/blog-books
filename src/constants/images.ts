export default {
  LOGO_URL: '/images/logo.png',
  LOGO_ALT: '49th Shelf',
};
