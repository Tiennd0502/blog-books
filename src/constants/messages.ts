export default {
  NO_DATA: 'No data',
  BACK_TO_HOME: 'Back to Home page',
  NOT_FOUND_PAGE: '404 - Page Not Found',
  DEFAULT_ERROR_APIS: 'Oops!!! Something went wrong!. Please try again later!',
};
