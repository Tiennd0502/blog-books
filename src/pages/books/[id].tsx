import React, { useState } from 'react';
import Image from 'next/image';
import { useRouter } from 'next/router';
import { GetStaticProps, GetStaticPaths, NextPage } from 'next';
import {
  Box,
  Heading,
  Text,
  List,
  Flex,
  ListItem,
  Button,
} from '@chakra-ui/react';
import { ParsedUrlQuery } from 'querystring';

import MainLayout from '@components/layouts/MainLayout';
import Loading from '@components/Loading';
import SEO from '@components/SEO';

import { Book } from '@interfaces/book';
import ENDPOINT from '@constants/apis';
import imageLoader from '@helpers/image';
import { getById, getAll, update } from '@services/mockApi';

interface BookPathParams extends ParsedUrlQuery {
  id: string;
}

interface BookPaths {
  params: BookPathParams;
}

type Props = {
  book: Book;
  error?: string;
};

export const getStaticPaths: GetStaticPaths = async () => {
  const { data: books } = await getAll<Book[]>(ENDPOINT.BOOK_URL);

  let paths: BookPaths[] = [];

  if (books) {
    paths = books.map((book) => {
      return {
        params: { id: book.id },
      };
    });
  }

  return { paths, fallback: true };
};

export const getStaticProps: GetStaticProps = async ({ params }) => {
  const { id } = params as BookPathParams;
  const { data: book, error } = await getById<Book>(ENDPOINT.BOOK_URL, id);

  if (error) {
    return {
      props: {
        error,
      },
    };
  }

  return {
    props: {
      book,
    },
    revalidate: 100,
  };
};

const BookDetail: NextPage<Props> = ({ book, error }) => {
  const { id, image, title, liked, summary, author, contributors } = book;
  const router = useRouter();
  const [statusLike, setStatusLike] = useState(liked);
  const [isLoading, setIsLoading] = useState(false);

  if (router.isFallback) {
    return <Loading />;
  }

  if (error) {
    // TODO: Define component for display error
    return (
      <MainLayout>
        <Heading m="0 auto" color="red.500" textAlign="center">
          {error}
        </Heading>
      </MainLayout>
    );
  }

  const handleChangeStatus = async (like: boolean) => {
    setIsLoading(true);
    const newBook = {
      ...book,
      liked: like,
    };
    await update<Book>(ENDPOINT.BOOK_URL, id, newBook);
    setStatusLike(like);
    setIsLoading(false);
  };

  return (
    <>
      <SEO
        title={title}
        siteTitle="New Releases · 49th Shelf"
        description={`${title}, ${author}, New Poetry · New Releases · 49th Shelf`}
      />
      <MainLayout>
        <Flex gap="10">
          <Box minWidth="30%">
            <Image
              src={image}
              alt={title}
              placeholder="blur"
              blurDataURL={`/_next/image?url=${image}&w=16&q=1`}
              layout="responsive"
              objectFit="cover"
              width="402"
              height="600"
              loader={imageLoader}
            />
            {isLoading && <Loading />}
            <Button
              textTransform="uppercase"
              display="block"
              w="100%"
              my="20px"
              bg={statusLike ? 'orange.500' : 'blue.700'}
              variant={statusLike ? 'warning' : ''}
              onClick={() => handleChangeStatus(!statusLike)}
            >
              like{statusLike ? 'd' : ''}
            </Button>
          </Box>
          <Box width="70%">
            <Heading mb="34px">{title}</Heading>
            <Text fontWeight="bold">Summary:</Text>
            <Text>{summary}</Text>
            <Text fontWeight="bold" display="inline">
              Author:&nbsp;
            </Text>
            <Text display="inline">{author}</Text>
            <br />
            <Text fontWeight="bold">Contributors:</Text>
            <List spacing={2} pl="40px">
              {contributors?.map((contributor) => (
                <ListItem key={contributor}>
                  <Text>{contributor}</Text>
                </ListItem>
              ))}
            </List>
          </Box>
        </Flex>
      </MainLayout>
    </>
  );
};

export default BookDetail;
