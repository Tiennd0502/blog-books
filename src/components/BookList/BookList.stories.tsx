import React from 'react';
import Storybook from '@storybook/react';

import { BOOK_LIST } from '@__mocks__/constants/books';
import BookList from './index';

import '@styles/main.css';

export default {
  title: 'Components/BookList',
  component: BookList,
} as Storybook.ComponentMeta<typeof BookList>;

const Template: Storybook.ComponentStory<typeof BookList> = (args) => (
  <BookList {...args} />
);

export const Default = Template.bind({});
Default.args = {
  books: BOOK_LIST,
};
