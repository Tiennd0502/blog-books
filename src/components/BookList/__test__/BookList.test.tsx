import { render } from '@testing-library/react';

import { BOOK_LIST } from '@__mocks__/constants/books';
import BookList from '../index';

describe('BookList component', () => {
  test('Should render BookList component', () => {
    const { getByRole } = render(<BookList books={BOOK_LIST} />);
    const bookList = getByRole('list');

    expect(bookList).toBeInTheDocument();
  });

  test('Should render enough elements', () => {
    const { getByRole } = render(<BookList books={BOOK_LIST} />);
    const bookList = getByRole('list');

    expect(bookList.childNodes).toHaveLength(BOOK_LIST.length);
  });

  test('Should render empty elements', () => {
    const { getByRole } = render(<BookList books={[]} />);
    const bookList = getByRole('list');

    expect(bookList.childNodes).toHaveLength(0);
  });

  test('Matches snapshot', () => {
    const { container } = render(<BookList books={BOOK_LIST} />);

    expect(container).toMatchSnapshot();
  });
});
