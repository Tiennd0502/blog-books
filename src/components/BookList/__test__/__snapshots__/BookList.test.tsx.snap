// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList component Matches snapshot 1`] = `
<div>
  <div
    class="css-pth8pb"
    role="list"
  >
    <div
      class="book css-0"
      role="link"
    >
      <div
        class="bookImgThumb css-0"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 166.66666666666669% 0px 0px 0px;"
          />
          <img
            alt="The Threshold"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(/_next/image?url=https://49thshelf.com/var/ezflow_site/storage/images/books/t/the-threshold/9780374604271_cover/604130664-1-eng-CA/9780374604271_cover_wide_960.jpg&w=16&q=1);"
          />
          <noscript />
        </span>
      </div>
      <div
        class="css-0"
      >
        <p
          class="chakra-text bookTitle css-0"
        >
          The Threshold
        </p>
        <p
          class="chakra-text css-0"
        >
          by 
          Iman Mersal
        </p>
      </div>
    </div>
    <div
      class="book css-0"
      role="link"
    >
      <div
        class="bookImgThumb css-0"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 166.66666666666669% 0px 0px 0px;"
          />
          <img
            alt="HARROWINGS"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(/_next/image?url=https://49thshelf.com/var/ezflow_site/storage/images/books/h/harrowings/9781772014051_cover/629613818-1-eng-CA/9781772014051_cover_wide_960.jpg&w=16&q=1);"
          />
          <noscript />
        </span>
      </div>
      <div
        class="css-0"
      >
        <p
          class="chakra-text bookTitle css-0"
        >
          HARROWINGS
        </p>
        <p
          class="chakra-text css-0"
        >
          by 
          Cecily Nicholson
        </p>
      </div>
    </div>
    <div
      class="book css-0"
      role="link"
    >
      <div
        class="bookImgThumb css-0"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 166.66666666666669% 0px 0px 0px;"
          />
          <img
            alt="Openwork and Limestone"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(/_next/image?url=https://49thshelf.com/var/ezflow_site/storage/images/books/o/openwork-and-limestone/9781989466438_cover/617742024-1-eng-CA/9781989466438_cover_wide_960.jpg&w=16&q=1);"
          />
          <noscript />
        </span>
      </div>
      <div
        class="css-0"
      >
        <p
          class="chakra-text bookTitle css-0"
        >
          Openwork and Limestone
        </p>
        <p
          class="chakra-text css-0"
        >
          by 
          Frances Boyle
        </p>
      </div>
    </div>
    <div
      class="book css-0"
      role="link"
    >
      <div
        class="bookImgThumb css-0"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 166.66666666666669% 0px 0px 0px;"
          />
          <img
            alt="Forget-Sadness-Grass"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(/_next/image?url=https://49thshelf.com/var/ezflow_site/storage/images/books/f/forget-sadness-grass/9781553806684_cover/629622610-1-eng-CA/9781553806684_cover_wide_960.jpg&w=16&q=1);"
          />
          <noscript />
        </span>
      </div>
      <div
        class="css-0"
      >
        <p
          class="chakra-text bookTitle css-0"
        >
          Forget-Sadness-Grass
        </p>
        <p
          class="chakra-text css-0"
        >
          by 
          Anthony Di Nardo
        </p>
      </div>
    </div>
    <div
      class="book css-0"
      role="link"
    >
      <div
        class="bookImgThumb css-0"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 166.66666666666669% 0px 0px 0px;"
          />
          <img
            alt="The Oysters I Bring to Banquets"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(/_next/image?url=https://49thshelf.com/var/ezflow_site/storage/images/books/t/the-oysters-i-bring-to-banquets/9781771837101_cover/620754283-1-eng-CA/9781771837101_cover_wide_960.jpg&w=16&q=1);"
          />
          <noscript />
        </span>
      </div>
      <div
        class="css-0"
      >
        <p
          class="chakra-text bookTitle css-0"
        >
          The Oysters I Bring to Banquets
        </p>
        <p
          class="chakra-text css-0"
        >
          by 
          Gary Geddes
        </p>
      </div>
    </div>
    <div
      class="book css-0"
      role="link"
    >
      <div
        class="bookImgThumb css-0"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 166.66666666666669% 0px 0px 0px;"
          />
          <img
            alt="Dream of Me as Water"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(/_next/image?url=https://49thshelf.com/var/ezflow_site/storage/images/books/d/dream-of-me-as-water/9781990293184_cover/620592551-1-eng-CA/9781990293184_cover_wide_960.jpg&w=16&q=1);"
          />
          <noscript />
        </span>
      </div>
      <div
        class="css-0"
      >
        <p
          class="chakra-text bookTitle css-0"
        >
          Dream of Me as Water
        </p>
        <p
          class="chakra-text css-0"
        >
          by 
          David Ly
        </p>
      </div>
    </div>
  </div>
</div>
`;
